src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -40,6 +46,8 @@ internal sealed class VisualStudioDocumentNavigationService : ForegroundThreadAf
         {
             _serviceProvider = serviceProvider;
             _editorAdaptersFactoryService = editorAdaptersFactoryService;
+            var runningDocumentTable = (IVsRunningDocumentTable)serviceProvider.GetService(typeof(SVsRunningDocumentTable));
```

Comment:
This is now taking a UI thread affinity here. This should be moved to only be initialized when needed.

Reply:
This service is free threaded and can be retrieved in the constructor - I had checked before with Ryan when I wrote the listener
e.g. https://github.com/dotnet/roslyn/blob/4d0c2c0d6fc6eb4d301454ec2e158b63b090f098/src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs#L74